src/main/kotlin/com/mdrobnak/lalrpop/inspections/PrecedenceInspection.kt

```diff
@@ -0,0 +1,77 @@
+package com.mdrobnak.lalrpop.inspections
+
+import com.intellij.codeInspection.LocalInspectionTool
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.psi.PsiElementVisitor
+import com.mdrobnak.lalrpop.psi.LpAlternative
+import com.mdrobnak.lalrpop.psi.LpAlternatives
+import com.mdrobnak.lalrpop.psi.LpAnnotation
+import com.mdrobnak.lalrpop.psi.LpVisitor
+
+object PrecedenceInspection : LocalInspectionTool() {
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean): PsiElementVisitor {
+        return object : LpVisitor() {
+            override fun visitAlternatives(alternatives: LpAlternatives) {
+                val alternativesWithoutPrecedenceAnnotation = mutableListOf<LpAlternative>()
+                var prevPrecedenceAnnotation: LpAnnotation? = null
+                alternatives.alternativeList.forEach {
+                    val precedenceAnnotation =
+                        it.annotationList.find { annotation -> annotation.annotationName.text == "precedence" }
+                    val assocAnnotation =
+                        it.annotationList.find { annotation -> annotation.annotationName.text == "assoc" }
+
+                    if (precedenceAnnotation != null) {
+                        if (alternativesWithoutPrecedenceAnnotation.isNotEmpty()) {
+                            alternativesWithoutPrecedenceAnnotation.forEach { alternative ->
+                                holder.registerProblem(
```

Comment:
I think this would register the problem multiple times if there's more than one alternative with the precedence annotation following an alternative without the annotation.

It also looks like alternatives without the annotation which follow those that have it will not be flagged. Maybe the algorithm here needs to be reworked to fix these issues.

Reply:
> -We could avoid having to re-annotate each alternative, taking the convention that if an alternative is not annotated, it inherits the annotation of the last explicit annotation. Note that in that case, the default associativity is not recoverable though: once we started annotating associativity, there is no way to return to the default behavior. But we can still drop the default behavior, or give it a proper annotation.

Oh I thought this was already done and I kept it in mind when I wrote the code. Will fix